### PR TITLE
Feat: add book and label classes with methods

### PR DESCRIPTION
**Pull request by Hope1226**
- [x] Create a Book class in a separate .rb file.

- [x] Create a Label class with an association to the Item class (in a separate .rb file).

- [x] All Book class properties visible in the diagram should be defined and set up in the constructor method.

- [x] All Label class properties visible in the diagram should be defined and set up in the constructor method.

- [x] Implement methods:

- add_item method in the Label class
:heavy_check_mark:  should take an instance of the Item class as an input
:heavy_check_mark:  should add the input item to the collection of items
:heavy_check_mark:  should add self as a property of the item object (by using the correct setter from the item object)

- can_be_archived?() in the Book class
:heavy_check_mark:  should override the method from the parent class
:heavy_check_mark: should return true if parent's method returns true OR if cover_state equals to "bad"
        otherwise, it should return false

### CODE REVIEW

File: label.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Label
+  attr_accessor :title, :color
+
+  attr_reader :books, :id
+
+  def initialize(title, color)
+    @id = rand(1000),
+          @title = title,
+          @color = color,
+          @books = []
```
Comment (nicupop729) [inline label.rb:10]: [OPTIONAL]

- [ ] In order to have a good looking code, you can take into consideration to indent properly these three lines.

File: label.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  def add_book(book)
+    @books << book
+    book.label = self
+  end
```
Comment (nicupop729) [inline label.rb:16]: Correct me if I'm wrong, but as I understand from the project requirements, we need to set labels on music albums and on games as well, not only on books.

- [ ] Please update this file with specific methods so labels can be added to games and music albums as well.

**Review by nicupop729 — CHANGES_REQUESTED** [review]
Hi @Hope1226 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: label.rb (hunk 1 above)
Comment (Hope1226) [reply label.rb:10]: Hi @nicupop729, thanks for your feedback, I totally agree on this one, but `linters` wanted it to be this way :man_shrugging:

File: label.rb (hunk 2 above)
Comment (Hope1226) [reply label.rb]: @nicupop729, thanks for your bringing this up :100: above comment is addressed with a respect

**Review by nicupop729 — APPROVED** [review]
Hi @Hope1226 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**